Implement a basic test suite with rspec + factory_bot

jamiew opened the pull request:
This is far from complete, but I wanted to submit now to get some other eyes and feedback on this so far.

* Added rspec and factory_bot
* Basic factories
* Simple model tests: factories are valid, required associations are present, required fields are present
* Controller tests for `GET projects#index` and `GET projects#show` but nothing else yet

A good test suite will make it a lot easier for others to contribute if they don't have to worry about breaking anything.

Next steps

* Setup CI -- any preference? Circle, something else?
* Test everything else
* Cleanup the jank

sradu commented:
This is awesome. Will look over it tmr. @OrlandoLee @adrianthedev if you want to check it out please give it a go!

OrlandoLee commented on test/application_system_test_case.rb:
```diff
@@ -1,5 +0,0 @@
-require "test_helper"
-
-class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
```
hmm. I am not sure if we wanna remove this. I feel if we wanna add more tests, it would be really useful if the test can be integration tests. So that writing one test can catch many cases in the whole flow.

jamiew commented:
I switched from test::unit/minitest to rspec, so I also removed the `test/` scaffolds in favor of `spec/`. We can do system tests w/ rspec, I agree would be great to add

jamiew commented:
I've added more scaffold and TODO tests, and implemented tests for everything in `AdminController`

I've also gotten the build working on CircleCI: https://circleci.com/gh/jamiew/covid-volunteers/tree/rspec

I'd venture this is good enough to merge, since any tests are better than no tests.
